src/wp-includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,884 @@
+<?php
+/**
+ * WordPress Plugin Administration API: WP_Plugin_Dependencies class
+ *
+ * @package WordPress
+ * @subpackage Administration
+ * @since 6.4.0
+ */
+
+/**
+ * Core class for installing plugin dependencies.
+ *
+ * It is designed to add plugin dependencies as designated in the
+ * `Requires Plugins` header to a new view in the plugins install page.
+ */
+class WP_Plugin_Dependencies {
+
+	/**
+	 * Holds 'get_plugins()'.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	protected static $plugins = array();
+
+	/**
+	 * Holds plugin directory names to compare with cache.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	protected static $plugin_dirnames = array();
+
+	/**
+	 * Holds cached plugin directory names.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	protected static $plugin_dirnames_cache = array();
+
+	/**
+	 * Holds sanitized plugin dependency slugs.
+	 *
+	 * Keyed on the dependent plugin's filepath,
+	 * relative to the plugins directory.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	protected static $dependencies = array();
+
+	/**
+	 * Holds an array of sanitized plugin dependency slugs.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	protected static $dependency_slugs = array();
+
+	/**
+	 * Holds an array of dependent plugin slugs.
+	 *
+	 * Keyed on the dependent plugin's filepath,
+	 * relative to the plugins directory.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	protected static $dependent_slugs = array();
+
+	/**
+	 * Holds 'plugins_api()' data for plugin dependencies.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	protected static $dependency_api_data = array();
+
+	/**
+	 * Holds plugin dependency filepaths, relative to the plugins directory.
+	 *
+	 * Keyed on the dependency's slug.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	protected static $dependency_filepaths = array();
+
+	/**
+	 * Initializes by fetching plugin header and plugin API data,
+	 * and deactivating dependents with unmet dependencies.
+	 *
+	 * @since 6.4.0
+	 */
+	public static function initialize() {
+		self::read_dependencies_from_plugin_headers();
+		self::get_dependency_api_data();
+		self::deactivate_dependents_with_unmet_dependencies();
+	}
+
+	/**
+	 * Determines whether the plugin has plugins that depend on it.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $plugin_file The plugin's filepath, relative to the plugins directory.
+	 * @return bool Whether the plugin has plugins that depend on it.
+	 */
+	public static function has_dependents( $plugin_file ) {
+		$dependency_paths = self::get_dependency_filepaths();
+		return in_array( $plugin_file, $dependency_paths, true );
```

Comment:
No need to run the logic of `::get_dependency_filepaths()`. Let's just slugify it and check it against the dependency slugs.
```suggestion
		$slug = str_contains( $plugin_file, '/' ) ? dirname( $plugin_file ) : $plugin_file;
		return in_array( $slug, self::$dependency_slugs, true );
```